### DISCUSSION TITLE
Daemonize and get the PID

### DISCUSSION THREAD

**Discussion by o-alquimista** [brgl/libgpiod#80] [Q&A]
When you use `--daemonize`, the CLI tools don't tell you the PID of the created process. Sure, I could use something like `pidof`, but what if multiple instances of `gpioset` are running at the time? This is not doable. I need another approach. What do you think?

I'm using the CLI tools from a shell script.
libgpiod v2.1.2

**Comment by mhei** [comment]
My first proposal would be to add the command line parameter `-P | --pid-file <pid-file>` which tells the tool to create a pid file in /var/run in good old Unix fashion. That file could then be processed by your shell script and with different names you could easily run different daemons in parallel.

**Reply by o-alquimista** [reply]
```sh
gpioset GPIO18=1 &
PID=$!
```

I daemonized it myself. This seems to work :)

And I second your proposal. I wonder what's the purpose of `--daemonize` without the PID.

**Comment by o-alquimista** [comment]
It's been two days now. I have been trying to write a script — which will be run by cron — to read the GPU temperature and control the case fan of my Raspberry Pi using libgpiod.

The unspecified behavior when you're not "holding" a line creates complications, because you have to come up with some lock-file logic to _try_ to prevent multiple instances of your script running at the same time. You'll always have to be "holding" a value.

I just want to set a value and have it "stick" there. I don't want to manage PIDs and have long-running tasks. _I don't want to write a daemon for something this simple_. It only took me a few minutes to write such script using [gpioctl](https://man.openbsd.org/OpenBSD-7.5/gpioctl.8) on OpenBSD.
___
Update: issue #47 will address this.

**Reply by brgl** [reply]
> Update: issue #47 will address this.

It would be awesome if you could take the series and play with it to see if it's useful for whatever use-case you have.

**Comment by brgl** [comment]
I don't have anything against adding this functionality. I'll add it to the TODO as I have too much on my professional plate to take care of this right now. But feel free to add this functionality to gpioset if you can and send patches for review to linux-gpio.

**Comment by o-alquimista** [comment]
I've written a shell script to control the Raspberry Pi case fan using libgpiod. This uses a PID file to manage the running process.

https://gitlab.com/crimson.king/rpi-case-fan-libgpiod

Can't wait for the DBus interface to be released :)
